pyuvdata/uvflag/uvflag.py

```diff
@@ -1063,20 +1097,24 @@ def read(self, filename, history='', run_check=True, check_extra=True,
                 else:
                     self.Nfreqs = np.unique(self.freq_array).size
 
-                self.history = header['history'][()].decode("utf8")
+                self.history = _read_uvflag_string(header['history'], filename)
 
                 self.history += history
 
                 if not uvutils._check_history_version(self.history, self.pyuvdata_version_str):
                     self.history += self.pyuvdata_version_str
 
                 if 'label' in header.keys():
-                    self.label = header['label'][()].decode("utf8")
+                    self.label = _read_uvflag_string(header['label'], filename)
 
-                polarization_array = header['polarization_array'][()]
-                if isinstance(polarization_array[0], np.string_):
-                    polarization_array = np.asarray(polarization_array,
-                                                    dtype=np.str_)
+                polarization_array_type = header['polarization_array'][()].dtype.type
+                if polarization_array_type in (np.int, np.int32, np.int64):
+                    polarization_array = header['polarization_array'][()]
+                else:
+                    # we have a string-type polarization array from collapsing
+                    polarization_array = np.asarray(
+                        header['polarization_array'][()], dtype=str
```

Comment:
sorry for my ignorance, but why does this not need the `_read_uvflag_string` call?

Reply:
This one is different because it's saved as a length-1 array with `str` datatype, rather than a scalar string. I agree it's a little un-intuitive, but I think it's probably the best way to handle the fact that we still want it to be an array on the object.